### PR TITLE
Created a Static Landing (Home) Page

### LINKED ISSUE #106: Design and develop front-end code for a static landing page
As of now, upon loading the website in its current state, a user will be faced with a barren `index.html` page that does not tell them anything about our product to them, nor provides any easy navigation or access to other parts of the Gamified Learning platform. Needless to say, this doesn't exactly set the best impression on potential new consumers. We need some sort of page to greet users with, inform them what Gamified is, and convince them to use our services.

The task is to design a mockup and create frontend code for a landing page that users would be greeted with, upon visiting the website. The landing page should have an appealing, modern design to attract potential consumers while advertising our product and its features, yet it should still be in line with the design from our other pages. 

For some inspiration, you can find a couple of landing page examples/templates [here](https://unbounce.com/landing-page-examples/best-landing-page-examples/) and [here](https://www.figma.com/community/file/1127302394641561751/50-landing-page-designs?searchSessionId=lsi3oza5-lisq8usiqi).

### PR DESCRIPTION
**Pull request by angadk4**
Developed a static landing page which welcomes users, provides basic information about the platform, and prompts them to begin now by registering.

**Changes Made:**
- Implemented all relevant frontend code for the landing page via **home.html**, **home.css** and **home.js**. 
- Edited the base (navigation bar) to now redirect to the new home page upon pressing _home_.
- Modified **app.py** to include new home route, and ensure index redirects to home. This way, users will be met with the landing page directly after loading up the website.
- Added a button at the bottom of the home page redirecting users to the registration.

**Future Changes:**
- Anyone with new ideas can add on/modify as needed.

### CODE REVIEW

**Comment by zareentk** [comment]
Primary Reviewer: @clarawong20

File: app/src/static/css/home.css [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+/**
+ * ========
+ * Variables
+ * ========
+**/
+
+:root {
+    /* Main Colours */
+    --box-clr: rgba(222, 221, 221, 0.3);
+    --heading-clr: #5b83eb;
```
Comment (clarawong20) [inline app/src/static/css/home.css:40]: add: --heading-dark-clr: #00245e; or any darker navy blue colour/black

File: app/src/static/css/home.css [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+
+#title-flex-container, #title-description-container, .title-image, #title, #description {
+    position: relative; /* This ensures that these elements are rendered above the pseudo-element */
+    z-index: 1;
+}
+
+#title {
+    font: var(--heading-font);
+    color: var(--heading-clr);
```
Comment (clarawong20) [inline app/src/static/css/home.css]: change:
 - color: var(--heading-dark-clr);

File: app/src/static/css/home.css [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+}
+
+#get-started-text {
+    font: var(--heading-font);
+    font-size: 30px;
+    margin-bottom: 5px;
+    text-align: center;
+    color: var(--heading-clr);
+    opacity: 80%;
+    text-decoration: underline;
```
Comment (clarawong20) [inline app/src/static/css/home.css]: get rid of the underline, looks better visually and less misleading

**Review by clarawong20 — COMMENTED** [review]
I looked over the home page and it looks really good! I added some comments and I'm just being picky with the visuals. For the "Welcome to our gamified learning platform" title, I think having a darker colour for the font looks better visually than a lighter font with black lining it, so I suggested a navy blue colour (but you can change it to whatever darker colour you think looks good). For the 'get started by clicking below' part, I just suggested removing the underline so a) it looks nicer and b) it's not misleading because I initially thought it was a link and tried clicking the text first instead of the image (which is really cool by the way). Other than that, great work!

**Comment by angadk4** [comment]
> I looked over the home page and it looks really good! I added some comments and I'm just being picky with the visuals. For the "Welcome to our gamified learning platform" title, I think having a darker colour for the font looks better visually than a lighter font with black lining it, so I suggested a navy blue colour (but you can change it to whatever darker colour you think looks good). For the 'get started by clicking below' part, I just suggested removing the underline so a) it looks nicer and b) it's not misleading because I initially thought it was a link and tried clicking the text first instead of the image (which is really cool by the way). Other than that, great work!

Sounds good, I've made the changes you recommended.

**Review by zareentk — APPROVED** [review]
LGTM! thanks Angad

**Review by clarawong20 — APPROVED** [review]

**Pull request merged**